src/Sarif/VersionOne/Readers/AnnotatedCodeLocationIdConverter.cs

```diff
@@ -29,7 +29,7 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             // when the input is a string that parses as a valid integer, such as "1". In
             // that case, objectType is Int32, but the object returned by reader.Value is
             // a string!
-            var stringValue = value as string;
+            string stringValue = value as string;
```

Comment:
>string [](start = 12, length = 6)

Huh, that answers I question I had in the editorconfig file. The `as` operator does _not_ make the type "apparent", and it wants an explicit type here (where I'd've been happy with `var`). No matter, it's fine.

Reply:
Sorry I was testing to see if the suggestion to use var shows up by replacing it with explicit type on the left side. I'll revert this.

---
In reply to: [388500810](https://github.com/microsoft/sarif-sdk/pull/1797#discussion_r388500810) [](ancestors = 388500810)

Reply:
Looks like as operator is not treated as obvious. 

I was testing var usage with the as operator and at some point, I thought the IDE didn't complain and had made the assumption that var usage is ok with the as operator, but looks like that's not the case.

---
In reply to: [388500810](https://github.com/microsoft/sarif-sdk/pull/1797#discussion_r388500810) [](ancestors = 388500810)

Reply:
No worries. We can zap most 'as' usage anyway with something like.

if (value is string stringValue)

---
In reply to: [389063699](https://github.com/microsoft/sarif-sdk/pull/1797#discussion_r389063699) [](ancestors = 389063699,388500810)